Sources/MongoSwift/WriteConcern.swift

```diff
@@ -131,17 +122,42 @@ public class WriteConcern: Codable {
         }
     }
 
-    /// Initializes a new `WriteConcern` by copying a `mongoc_write_concern_t`.
+    /// Initializes a new `WriteConcern` with the same values as the provided `mongoc_write_concern_t`.
     /// The caller is responsible for freeing the original `mongoc_write_concern_t`.
     internal init(from writeConcern: OpaquePointer?) {
-        self._writeConcern = mongoc_write_concern_copy(writeConcern)
+        if mongoc_write_concern_journal_is_set(writeConcern) {
+            self.journal = mongoc_write_concern_get_journal(writeConcern)
+        } else {
+            self.journal = nil
+        }
+
+       let number = mongoc_write_concern_get_w(writeConcern)
+        switch number {
+        case MONGOC_WRITE_CONCERN_W_DEFAULT:
+            self.w = nil
+        case MONGOC_WRITE_CONCERN_W_MAJORITY:
+            self.w = .majority
+        case MONGOC_WRITE_CONCERN_W_TAG:
+            if let wTag = mongoc_write_concern_get_wtag(writeConcern) {
+                self.w = .tag(String(cString: wTag))
+            }
+        default:
+            self.w = .number(number)
+        }
+
+       let wtimeout = Int64(mongoc_write_concern_get_wtimeout(writeConcern))
+       if wtimeout == 0 {
```

Comment:
similar to comment above, do we need to explicitly set it to nil or can we have an `if` that sets the property when it's non-zero?

Reply:
If we make the journal, w, and wtimeoutMS immutable as @patrickfreed suggests, Swift will complain about the values not being initialized because there's no `else`.